Add RoundRobinLookupStrategy and change other strategies

SlavoKrupa opened the pull request:
* When no service was found exception message had no added information.
"IllegalArgumentException:bound must be positive"
or propagated NullPointer exception.
Now runtime exception explicitly mentions metadata without
microservice implementation.


* New implemenation of lookup factory based on Round-robin scheduling is
implemented.

* Default lookup strategy is changed to Round-robin.

livthomas commented on microservices/src/main/java/io/silverware/microservices/silver/services/lookup/RoundRobinLookupStrategy.java:
```diff
@@ -0,0 +1,46 @@
+      if (Integer.MAX_VALUE == index) {
+         index = 0;
+      }
+      return servicesArray[index % servicesArray.length];
```
I think this method can be called from different threads at the same time which can now cause problems (for example `ArrayIndexOutOfBoundsException` on line 40). At least the part where the counter is increased (lines 40-43) should be synchronized. Or maybe also line 44 should be included. @marvec, what do you think?

SlavoKrupa committed:
Add RoundRobinLookupStrategy and change other strategies


* When no service was found exception message had no added information.
"IllegalArgumentException:bound must be positive"
or propagated NullPointer exception.
Now runtime exception explicitly mentions metadata without
microservice implementation.


* New implemenation of lookup factory based on Round-robin scheduling is
implemented.

* Default lookup strategy is changed to Round-robin.

* Implementation covered by tests

* Upgraded jmockit version
